packages/agents/src/agents/utils/Agent.ts

```diff
@@ -38,19 +40,11 @@ export class Agent<TRunArgs = GoalRunArgs> implements RunnableAgent<TRunArgs> {
   public async* run(
     args: TRunArgs,
   ): AsyncGenerator<AgentOutput, RunResult, string | undefined> {
-    return yield* this.runWithChat([
-      // Add an extra prompt informing agent about variable usage
-      {
-        role: "system",
-        content: `Variables are annotated using the \${variable-name} syntax. Variables can be used as function argument using the \${variable-name} syntax. Variables are created as needed, and do not exist unless otherwise stated.`
-      },
-      ...this.config.prompts.initialMessages(args)
-    ]);
+    this.initializeChat(args);
+    return yield* this.runWithChat();
   }
 
-  public async* runWithChat(
-    messages: ChatMessage[],
-  ): AsyncGenerator<AgentOutput, RunResult, string | undefined> {
```

Comment:
I'd still keep this and the the empty args signature as an overload. For easier testing, as it's more DI

Reply:
Sounds good, I'll do that in a separate PR